### DISCUSSION TITLE
Prepare release asset ... issue

### DISCUSSION THREAD

**Discussion by saniho** [saniho/apiEnedis#149] [General]
@mdeweerd 

premier test de l'action "Prepare release asset ", mais plantage : 

(https://github.com/saniho/apiEnedis/runs/5529583975?check_suite_focus=true#logs)"

**Comment by mdeweerd** [comment]
La poule et l'oeuf: le release interactif pose le tag dans lequel il faut mettre la version ce qui ne fonctionne que si on travaille dans une branche...

J'ai trouvé une action qui permet de modifier du json et du yaml mais pas le python.
Du coup une évoution va désormais lire la version depuis le fichier 'manifest.json' avec un code que j'ai écrit et porté ici.

J'ai fait une pré-release.

Notons que le tag posé commence avec la lettre 'v', ce que j'observe généralement - je l'ai probablement fait par automatisme.

La pre-release a fonctionné: le manifest a été mis à jour, le tag déplacé et le zip créé avec le contenu voulu.

**Reply by mdeweerd** [reply]
Le code prenant en compte le manifest n'avait pas été propagée - c'est fait (il y a aussi un test pour vérifier que la version donné sous python et bien celui du manifest).

Normalement les versions devraient juste avoir 3 nombres, majeur, mineur et patch. On peut définir un autre tag, mais c'est moins automatique.

Il y a donc la v1.5.0.3 en pre-release permettant de regarder le contenu du zip et le tag. Le manifest a été mis à jour par l'action ;-).

Comme il y a desormais un zip, il est aussi possible de mettre "zip_release" à true dans le hacs.json .

**Reply by mdeweerd** [reply]
Autre chose, actuellement l'expression régulière pour reconnaître la version est d"fini comme ceci dans release.yaml:
```yaml
version-regexp: \d+\.\d+\.\d+\.\d+
```

Donc si on passe à X.Y.Z, il faudra ajuster cela également.

**Comment by saniho** [comment]
j"ai intégré la derniere release sur mon ha, cela à bien fonctionné, je te propose qu'on livre en prod, je te laisse faire ta premiere version de prod ;)

**Reply by mdeweerd** [reply]
Ok, d'accord pour le passer version ZIP (modification de hacs.json à faire)?

**Reply by mdeweerd** [reply]
J'ai fait le release, l'officialisation du zip dans HACS sera pour une prochaine fois.
Comme la dernière publication "publique" état la 1.3.2.7, j'ai essayé de résumé un peu la description de la release.

**Comment by mdeweerd** [comment]
J'ai vu que l'action de release n'a pas réussi dans dev à cause de la limitation sur le push.

Il y a probablement 2 possibilités:
1. Autoriser le push au CI;
2. Faire les release sous main (il faudrait plutôt protéger main que dev je pense, ce qui nécessite une exception de push pour CI sur le main).

(Je pense avoir vu qu'on pouvait faire une exception sur la protections des push sur main).

Il faudra avoir le zip car j'ai indiqué que la release est fait sous forme de ZIP qui manque actuellement).

**Reply by mdeweerd** [reply]
Peut-être qu'en autorisant le "force push", le déplacement du tag devient possible en CI:

https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/about-protected-branches#allow-force-pushes

**Reply by saniho** [reply]
En effet cela semble etre une solution...

**Comment by saniho** [comment]
hello @mdeweerd , j'ai fait une nouvelle release pour corrige un petit bug, mais de mémoire, cela aurait du generer la release ... mais en fait il semble que cela n'est pas fonctionné ... j'ai peut-etre oublié untruc, tellement cela fait longtemps que je n'y ai pas touché
